compiler-plugin-tests/src/test/java/io/ballerina/stdlib/rabbitmq/plugin/RabbitmqCompilerPluginTest.java

```diff
@@ -272,19 +298,84 @@ public void testInvalidService15() {
-        assertDiagnostic(diagnostic, PluginConstants.CompilationErrors.INVALID_FUNCTION);
+        assertDiagnostic(diagnostic, INVALID_FUNCTION);
     }
 
     @Test
     public void testInvalidService16() {
         Package currentPackage = loadPackage("invalid_service_16");
         PackageCompilation compilation = currentPackage.getCompilation();
         DiagnosticResult diagnosticResult = compilation.diagnosticResult();
-        Assert.assertEquals(diagnosticResult.errors().size(), 3);
+        Assert.assertEquals(diagnosticResult.errors().size(), 2);
+        Object[] diagnostics = diagnosticResult.errors().toArray();
+        for (Object obj : diagnostics) {
+            Diagnostic diagnostic = (Diagnostic) obj;
+            assertDiagnostic(diagnostic, INVALID_FUNCTION_PARAM_MESSAGE_OR_PAYLOAD);
+        }
+    }
+
+    @Test
+    public void testInvalidService17() {
+        Package currentPackage = loadPackage("invalid_service_17");
+        PackageCompilation compilation = currentPackage.getCompilation();
+        DiagnosticResult diagnosticResult = compilation.diagnosticResult();
+        Assert.assertEquals(diagnosticResult.errors().size(), 2);
+        Object[] diagnostics = diagnosticResult.errors().toArray();
+        for (Object obj : diagnostics) {
+            Diagnostic diagnostic = (Diagnostic) obj;
+            assertDiagnostic(diagnostic, INVALID_FUNCTION_PARAM_MESSAGE_OR_CALLER);
+        }
+    }
+
+    @Test
+    public void testInvalidService18() {
+        Package currentPackage = loadPackage("invalid_service_18");
+        PackageCompilation compilation = currentPackage.getCompilation();
+        DiagnosticResult diagnosticResult = compilation.diagnosticResult();
+        Assert.assertEquals(diagnosticResult.errors().size(), 2);
+        Object[] diagnostics = diagnosticResult.errors().toArray();
+        for (Object obj : diagnostics) {
+            Diagnostic diagnostic = (Diagnostic) obj;
+            assertDiagnostic(diagnostic, INVALID_FUNCTION_PARAM_CALLER);
+        }
+    }
+
+    @Test
+    public void testInvalidService19() {
+        Package currentPackage = loadPackage("invalid_service_19");
+        PackageCompilation compilation = currentPackage.getCompilation();
+        DiagnosticResult diagnosticResult = compilation.diagnosticResult();
+        Assert.assertEquals(diagnosticResult.errors().size(), 2);
+        Object[] diagnostics = diagnosticResult.errors().toArray();
+        for (Object obj : diagnostics) {
+            Diagnostic diagnostic = (Diagnostic) obj;
+            assertDiagnostic(diagnostic, INVALID_FUNCTION_PARAM_PAYLOAD);
+        }
+    }
+
+    @Test
+    public void testInvalidService20() {
+        Package currentPackage = loadPackage("invalid_service_20");
+        PackageCompilation compilation = currentPackage.getCompilation();
+        DiagnosticResult diagnosticResult = compilation.diagnosticResult();
+        Assert.assertEquals(diagnosticResult.errors().size(), 2);
```

Comment:
```suggestion
        Assert.assertEquals(diagnosticResult.errorCount(), 2);
```

Reply:
Updated in https://github.com/ballerina-platform/module-ballerinax-rabbitmq/pull/393/commits/b981317fec0692ca8130a099798591997bf0bba1